GaloyMoney/galoy
# refactoring: remove some dependency on non typed function in ledger

ledger/query.ts is code copied/paste from core without type function.
remove the one that have no longer any reasons to be here

remove debug logger

removing csv account export out of core

chore: improve naming and error management

refactor: migrate admin requests to apps (#763)

* feat: make it easier to understand the diff of a .graphql

by computing another generated file which is lexicographic based
to remove the non deterministic aspect of the .graphql file creation

still keeping the standard schema.graphql as this is the one
that make most sense for an API user

* migrate admin use cases

* moving coordinate -> coordinate across codebase

also type improvement
+ validation for business coordinates/title input
+ migration

* splitting user/account/wallet for graphql admin

* moving isTestAccountPhone from app to domain

* need to remove type to avoid collision between Account in admin and root

* some small fix/naming improvment

* improve typing and boundaries between User and Account

* refactor: using more injection dependancy like pattern

* fix checkedToUsername casting

* fix defaultWalletId / publicWalletId mixed up

* remove resolver, adding description

* (temp) fix type mix between WalletId and WalletPublicId

* narrow down error type

* findByPhone no longer needed

* fix testAccount singular/plural mismatch

* misc improvment

* validate AccountStatus

* Update src/graphql/error-map.ts

Co-authored-by: Arvin <17693119+vindard@users.noreply.github.com>

* Update src/app/accounts/update-account-level.ts

Co-authored-by: Arvin <17693119+vindard@users.noreply.github.com>

* Update src/services/mongoose/accounts.ts

Co-authored-by: Arvin <17693119+vindard@users.noreply.github.com>

* Update src/graphql/admin/root/mutation/business-update-map-info.ts

Co-authored-by: Arvin <17693119+vindard@users.noreply.github.com>

* Update src/graphql/admin/root/mutation/account-update-status.ts

Co-authored-by: Arvin <17693119+vindard@users.noreply.github.com>

* Update src/graphql/root/query/user-default-wallet-id.ts

Co-authored-by: Arvin <17693119+vindard@users.noreply.github.com>

* missing import

* move TestAccounts to TestAccountsChecker

also fix the directory from user to account

* fixing shield path

* some naming improvment

* remove some throw from graphql

* Revert "remove some throw from graphql"

This reverts commit 1717e52e58f6506ad091b319a0ea82438f3c4713.

* some missing rename

* fix important error when rate limit is being hit

* create an editor role by default when running the tests

Co-authored-by: Arvin <17693119+vindard@users.noreply.github.com>

fix all the tests

add the migration script

feat: make it easier to understand the diff of a .graphql

by computing another generated file which is lexicographic based
to remove the non deterministic aspect of the .graphql file creation

still keeping the standard schema.graphql as this is the one
that make most sense for an API user

migrate admin use cases

moving coordinate -> coordinate across codebase

also type improvement
+ validation for business coordinates/title input
+ migration

splitting user/account/wallet for graphql admin

moving isTestAccountPhone from app to domain

need to remove type to avoid collision between Account in admin and root

some small fix/naming improvment

refactor: using more injection dependancy like pattern

fix defaultWalletId / publicWalletId mixed up

(temp) fix type mix between WalletId and WalletPublicId

narrow down error type

fix testAccount singular/plural mismatch

misc improvment

Update src/graphql/admin/root/mutation/account-update-status.ts

Co-authored-by: Arvin <17693119+vindard@users.noreply.github.com>

missing import

move TestAccounts to TestAccountsChecker

also fix the directory from user to account

some naming improvment

remove some throw from graphql

Revert "remove some throw from graphql"

This reverts commit 1717e52e58f6506ad091b319a0ea82438f3c4713.

migrate admin use cases

splitting user/account/wallet for graphql admin

migrate admin use cases

complete addEarn more-efficient use case

improve some types

moving getHash and LoggedError out of core/utils

move btc2sat and sat2btc out of core/utils

moving the remaining of utils out of core

removing last reference of core on graphql/
